### PR TITLE
basemaps: add missing tests

### PR DESCRIPTION
**Pull request by matallo**
found some issue while working on basemaps, these missing tests should cover current functionality with basemap content view:

- header
- source
- select

for new layers, before changing baselayer, it is created without category so it failed, fixed now

CR @xavijam /cc @javisantana

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ module.exports = cdb.core.View.extend({
           name: m.name,
           className: m.className,
           attribution: m.attribution,
-          source: source,
+          category: source,
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js]: Why do we have source but we call it category?

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ module.exports = cdb.core.View.extend({
     this._sourcesCollection = new CarouselCollection(
       _.map(this._getSources(), function (source) {
         return {
-          selected: this._baseLayer.get('category') === source,
+          selected: this._getCategory() === source,
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js]: This is a bit confusing, what is category and what is source?

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js [hunk 3]
```diff
@@ -65,7 +65,8 @@ module.exports = cdb.core.View.extend({
       }, this)
     );
 
-    this._sourcesCollection.bind('change:selected', this._renderBasemapSelect, this);
+    this._sourcesCollection.bind('change:selected', this._renderSelect, this);
+    this._baseLayer.bind('change', this._renderHeader, this);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js]: I guess you need to `add_related_model` both models/collections.

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header-view.js [hunk 4]
```diff
@@ -3,11 +3,17 @@ var template = require('./basemap-header.tpl');
 
 module.exports = cdb.core.View.extend({
 
+  initialize: function (opts) {
+    if (!opts.category) throw new Error('category is required');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header-view.js]: What is category? Model? Collection? simple?

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header.tpl [hunk 5]
```diff
@@ -6,6 +6,6 @@
     <div class="Editor-HeaderInfo-title u-bSpace">
       <h2 class="Editor-HeaderInfo-titleText CDB-Text CDB-Size-huge is-light u-ellipsis"><%- title %></h2>
     </div>
-    <p class="CDB-Text CDB-Size-medium u-altTextColor u-iBlock"><%- description %></p>
+    <p class="Editor-HeaderInfo-description CDB-Text CDB-Size-medium u-altTextColor u-iBlock"><%- description %></p>
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header.tpl]: We should review those classes @piensaenpixel, didn't we change them to `Editor-headerInfo`?

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-select-view.js [hunk 6]
```diff
@@ -46,6 +45,7 @@ module.exports = cdb.core.View.extend({
     this.clearSubViews();
     this.$el.html(template());
 
+    this._initCollection();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-select-view.js]: Let's not change/create/modify any model/collection in a render function, super important.

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js (hunk 1 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js]: it is a source indeed, but we need to call it category for backwards compatibility https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/table/table.js#L109

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js]: If we can't change it, why not keep calling it category everywhere?

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header-view.js (hunk 4 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header-view.js]: it's just a string

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js (hunk 1 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js]: cool, just following mockup nomenclature, I'm not inclined to a specific name, changing it then

**Comment by matallo** [comment]
💚   /cc @xavijam

File: lib/assets/javascripts/cartodb3/components/mosaic/mosaic.tpl [hunk 7]
```diff
@@ -1 +1 @@
-<ul class="Mosaic-list js-mosaic"></ul>
+<ul class="Mosaic-list js-list"></ul>
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/mosaic/mosaic.tpl]: Mosaic! yeah!

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js [hunk 8]
```diff
@@ -25,10 +26,12 @@ module.exports = cdb.core.View.extend({
   },
 
   _initCollections: function () {
+    var self = this;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js]: I feel we don't need the self variable.

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js [hunk 9]
```diff
@@ -52,82 +55,113 @@ module.exports = cdb.core.View.extend({
...
         };
       }, this)
     );
 
-    this._sourcesCollection.bind('change:selected', this._renderBasemapSelect, this);
+    this._categoriesCollection.bind('change:selected', this._renderSelect, this);
+    this.add_related_model(this._categoriesCollection);
+
+    this._baseLayer.bind('change', function () {
+      self._renderHeader();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js]: `this` is valid, you have chosen that context in the last parameter ;).
`self` 🔥 .

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js [hunk 10]
```diff
@@ -52,82 +55,113 @@ module.exports = cdb.core.View.extend({
...
+      return categoryMdl.get('selected');
     }, this);
 
-    return selectedSource.get('val');
+    return selectedCategory.get('val');
+  },
+
+  _getBaseLayerCategory: function () {
+    // baseLayer has no category at map creation
+    var category = this._baseLayer.get('category') ? this._baseLayer.get('category') : this._getDefaultCategory();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-view.js]: `this._baseLayer.get('category') || this._getDefaultCategory();` ?

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header-view.js (hunk 4 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header-view.js]: :+1:

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header-view.js [hunk 11]
```diff
@@ -3,11 +3,17 @@ var template = require('./basemap-header.tpl');
...
+
+    this.category = opts.category;
+  },
+
   render: function () {
     this.$el.html(
       template({
         title: _t('editor.layers.basemap.title-label'),
-        description: this.model.getName() + ' by CartoDB'
+        description: this.model.getName() + ' by ' + this.category
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header-view.js]: Let's use internationalization strings, 🙏 .

File: lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header.tpl (hunk 5 above)
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/basemap-content-views/basemap-header.tpl]: ping @piensaenpixel

**Comment by xavijam** [comment]
👍 minor things.

**Comment by matallo** [comment]
retest this, please

**Comment by matallo** [comment]
🚀

**Pull request merged**